samples/nrf_desktop/CMakeLists.txt

```diff
@@ -72,75 +70,7 @@ add_subdirectory(src/hw_interface)
 add_subdirectory(src/modules)
 add_subdirectory(src/services)
 
-if (${CONFIG_BOOTLOADER_MCUBOOT})
-  set(MCUBOOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../mcuboot)
-
-  if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/mcuboot_${CMAKE_BUILD_TYPE}.conf")
-    message(FATAL_ERROR "Configuration ${BOARD}/mcuboot_${CMAKE_BUILD_TYPE}.conf must be specified")
-  endif()
-
-  if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/dts.overlay")
-    message(FATAL_ERROR "Overlay ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/dts.overlay must be specified")
-  endif()
-
-  if (${CONFIG_DESKTOP_SIGNATURE_KEY_FILE})
-    set(key_file_path ${CONFIG_DESKTOP_SIGNATURE_KEY_FILE_PATH})
-  else()
-    message(WARNING "Using default key file from MCUBoot")
-    set(key_file_path "${MCUBOOT_DIR}/root-rsa-2048.pem")
-  endif()
-
-  set(cmd_sign_hex ${CMAKE_COMMAND} -E env
-      ${MCUBOOT_DIR}/scripts/imgtool.py sign
-      --key ${key_file_path}
-      --header-size ${CONFIG_TEXT_SECTION_OFFSET}
-      --align 8
-      --version ${CMAKE_PROJECT_VERSION}
-      --slot-size ${CONFIG_FLASH_LOAD_SIZE}
-      ${APPLICATION_BINARY_DIR}/zephyr/zephyr.hex
-      ${APPLICATION_BINARY_DIR}/zephyr/zephyr_signed.hex
-      DEPENDS ${APPLICATION_BINARY_DIR}/zephyr/zephyr.elf
-  )
-
-  set(cmd_sign_bin ${CMAKE_COMMAND} -E env
-      ${MCUBOOT_DIR}/scripts/imgtool.py sign
-      --key ${key_file_path}
-      --header-size ${CONFIG_TEXT_SECTION_OFFSET}
-      --align 8
-      --version ${CMAKE_PROJECT_VERSION}
-      --slot-size ${CONFIG_FLASH_LOAD_SIZE}
-      ${APPLICATION_BINARY_DIR}/zephyr/zephyr.bin
-      ${APPLICATION_BINARY_DIR}/zephyr/zephyr_signed.bin
-      DEPENDS ${APPLICATION_BINARY_DIR}/zephyr/zephyr.elf
-  )
-
-  add_custom_target(sign
-    COMMAND
-    ${cmd_sign_hex}
-    COMMAND
-    ${cmd_sign_bin}
-    COMMENT
-    "Sign binary files"
-    USES_TERMINAL
-    )
-  set(BOARD_ROOT_STR "")
-  foreach(root ${BOARD_ROOT})
-    set(BOARD_ROOT_STR "${BOARD_ROOT_STR}\;${root}")
-  endforeach()
-
-  add_custom_target(mcuboot
-    COMMAND
-    rm -rf mcuboot && mkdir mcuboot && cd mcuboot
-    COMMAND
-    cmake ${MCUBOOT_DIR}/boot/zephyr/
-    -G${CMAKE_GENERATOR}
-    -DBOARD=${BOARD}
-    -DBOARD_ROOT="${BOARD_ROOT_STR}"
-    -DDTC_OVERLAY_FILE="${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/flash.overlay"
-    -DCONF_FILE="${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/mcuboot_${CMAKE_BUILD_TYPE}.conf"
-    COMMENT
-    "Generating bootloader for application in mcuboot folder"
-    USES_TERMINAL
-    )
-
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  assert_exists(mcuboot_CONF_FILE)
```

Comment:
file existence is asserted but files were removed

Reply:
This file, and others of the same type, have not been removed: 

https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/683/files#diff-009e72229c5dff9599973130b6594a5f

Reply:
I tried to build the project from the branch and there are no files.
`samples/nrf_desktop/configuration/nrf52840_pca10059/flash.overlay` is marked as deleted in PR changes

Reply:
But I did accidentally forget to remove the assert on mcuboot_DTC_OVERLAY_FILE. Fixed now.

Reply:
Thanks